File: problems/hard/morgan_and_string/__tests__/test_morgan_string.js
const morganAndString = require("../morgan_and_string");

function runTests() {
    if (morganAndString("ACA", "BCF") !== "ABCACF") {
        throw new Error("Test 1 failed");
    }
    if (morganAndString("JACK", "DANIEL") !== "DAJACKNIEL") {
        throw new Error("Test 2 failed");
    }
    console.log("✅ All JavaScript tests passed!");
}

runTests();

File: problems/hard/morgan_and_string/__tests__/test_morgan_string.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include "../morgan_string.c"

void assert_equal(const char* result, const char* expected, const char* test_name) {
    if (strcmp(result, expected) != 0) {
        printf("❌ %s failed: expected %s, got %s\n", test_name, expected, result);
        exit(1);
    } else {
        printf("✅ %s passed\n", test_name);
    }
}

int main() {
    char* res1 = morganAndString("ACA", "BCF");
    assert_equal(res1, "ABCACF", "Test 1");
    free(res1);

    char* res2 = morganAndString("JACK", "DANIEL");
    assert_equal(res2, "DAJACKNIEL", "Test 2");
    free(res2);

    printf("🎉 All C tests passed!\n");
    return 0;
}
